docs/reference/inference/unified-inference.asciidoc

```diff
@@ -0,0 +1,411 @@
+[role="xpack"]
+[[unified-inference-api]]
+=== Unified inference API
+
+Streams a chat completion response using the Unified Schema format.
+
+IMPORTANT: The {infer} APIs enable you to use certain services, such as built-in {ml} models (ELSER, E5), models uploaded through Eland, Cohere, OpenAI, Azure, Google AI Studio, Google Vertex AI, Anthropic, Watsonx.ai, or Hugging Face.
+For built-in models and models uploaded through Eland, the {infer} APIs offer an alternative way to use and manage trained models.
+However, if you do not plan to use the {infer} APIs to use these models or if you want to use non-NLP models, use the <<ml-df-trained-models-apis>>.
+
+
+[discrete]
+[[unified-inference-api-request]]
+==== {api-request-title}
+
+`POST /_inference/<inference_id>/_unified`
+
+`POST /_inference/<task_type>/<inference_id>/_unified`
+
+
+[discrete]
+[[unified-inference-api-prereqs]]
+==== {api-prereq-title}
+
+* Requires the `monitor_inference` <<privileges-list-cluster,cluster privilege>>
+(the built-in `inference_admin` and `inference_user` roles grant this privilege)
+* You must use a client that supports streaming.
+
+
+[discrete]
+[[unified-inference-api-desc]]
+==== {api-description-title}
+
+The unified {infer} API enables real-time responses for completion tasks by delivering answers incrementally, reducing response times during computation.
+It only works with the `completion` task type for OpenAI and Elastic Inference Service. The Unified Schema defines a common schema for use across multiple services.
+
+
+[discrete]
+[[unified-inference-api-path-params]]
+==== {api-path-parms-title}
+
+`<inference_id>`::
+(Required, string)
+The unique identifier of the {infer} endpoint.
+
+
+`<task_type>`::
+(Optional, string)
+The type of {infer} task that the model performs. If included, this must be set to the value `completion`.
+
+
+[discrete]
+[[unified-inference-api-request-body]]
+==== {api-request-body-title}
+
+`messages`::
+(Required, array of objects) A list of objects representing the conversation.
++
+.Assistant message
+[%collapsible%closed]
+=====
+`content`::
+(Required unless `tool_calls` is specified, string or array of objects)
+The contents of the message.
++
+include::inference-shared.asciidoc[tag=unified-schema-content-with-examples]
```

Comment:
I personally find it helpful if examples are sprinkled throughout the docs as we're reading instead of just include complete ones at the bottom. I'm open to other suggestions here.

Reply:
This is probably a larger issue than this PR, but theres a bug with the example expansion where the arrow always points down ![Screenshot 2024-12-16 at 2 06 54 PM](https://github.com/user-attachments/assets/42472e51-b439-4232-941c-2d5e467e1731), but only for the inner expansion box, the outer one is correc.t

Reply:
Thanks for pointing that out. I made the docs team aware.